client/my-sites/domains/domain-management/components/header/domain-management-header.tsx

```diff
@@ -0,0 +1,155 @@
+import Breadcrumbs from 'calypso/my-sites/domains/domain-management/components/breadcrumbs';
+import { domainManagementList, isUnderDomainManagementAll } from 'calypso/my-sites/domains/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import isPrimaryDomainBySiteId from 'calypso/state/selectors/is-primary-domain-by-site-id';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { DomainManagementHeaderPassedProps, DomainManagementHeaderProps } from './types';
+import type { TranslateResult } from 'i18n-calypso';
+import './style.scss';
+
+const DomainManagementHeader = ( props: DomainManagementHeaderProps ) => {
+	const { selectedDomainName, isManagingAllDomains, selectedSite } = props;
+	const { __ } = useI18n();
+	let formattedHeaderText = selectedDomainName;
+	if ( ! selectedDomainName ) {
+		formattedHeaderText = isManagingAllDomains ? __( 'All Domains' ) : __( 'Site Domains' );
+	}
+
+	const renderCircle = () => (
+		<SVG viewBox="0 0 24 24">
+			<Circle cx="12" cy="12" r="12" />
+		</SVG>
+	);
+
+	const renderBreadcrumbs = () => {
+		const items = [
+			{
+				// translators: Internet domains, e.g. mygroovydomain.com
+				label: __( 'Domains' ),
+				href: domainManagementList( selectedSite!.slug, selectedDomainName ),
+			},
+			{
+				label: selectedDomainName,
+			},
+		];
+
+		const mobileItem = {
+			label: __( 'Back to domains' ),
+			href: domainManagementList( selectedSite!.slug, selectedDomainName ),
+			showBackArrow: true,
+		};
+
+		return <Breadcrumbs items={ items } mobileItem={ mobileItem } />;
+	};
+
+	const renderSuccessBadge = ( description: TranslateResult, icon?: JSX.Element ) => (
+		<Badge className="domain-management-header__badge domain-management-header__badge--success">
+			{ icon ? (
+				<Icon icon={ icon } size={ 14 } />
+			) : (
+				<div className="domain-management-header__badge-indicator">{ renderCircle() }</div>
+			) }
+			{ description }
+		</Badge>
+	);
+
+	const renderWarningBadge = ( description: TranslateResult ) => (
+		<Badge className="domain-management-header__badge domain-management-header__badge--success">
+			<div className="domain-management-header__badge-indicator">{ renderCircle() }</div>
+			{ description }
+		</Badge>
+	);
+
+	const renderPremiumBadge = () => (
+		<Badge className="domain-management-header__badge domain-management-header__badge--premium">
+			{ __( 'Premium domain' ) }
+			<Icon icon={ info } size={ 17 } />
+		</Badge>
+	);
+
+	const renderNeutralBadge = ( description: TranslateResult ) => (
+		<Badge className="domain-management-header__badge domain-management-header__badge--neutral">
+			{ description }
+		</Badge>
+	);
+
+	const renderTransferOrMappingBadge = ( type: string ) => {
+		if ( type === DomainType.MAPPED ) {
+			return renderNeutralBadge( __( 'Registered with an external provider' ) );
+		}
+
+		return renderNeutralBadge( __( 'Domain transfer' ) );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Domain Transfer' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-header)